internal/proxy/proxy.go

```diff
@@ -57,7 +67,13 @@ func (p Proxy) Apply(ctx context.Context, http, https, ftp, socks, no, mode stri
 		return err
 	}
 
-	return p.applyToEnvironment()
+	if err := p.applyToEnvironment(); err != nil {
+		return err
+	}
+	if err := p.applyToApt(); err != nil {
+		return err
+	}
```

Comment:
Tbh, I don't think it's out of scope since this PR is about adding the apt proxy configuration.
Also, you don't need to overcomplicate with Wait groups as well, I think it might be enough to join the errors and then returning them instead of returning at the first apply if it fails.
But if you feel like it's better to do it in another PR, that's fine, as long as it doesn't drag itself too much.